### PR TITLE
add fypp rules for measuring time

### PR DESCRIPTION
**Pull request by jvdp1**
Here is a procedure to measure time, using fypp pre-processing.

### CODE REVIEW

**Comment by wyphan** [comment]
LGTM. What variable type for `args` do you pass into `TOC`?

File: src/common/common.fypp [hunk 1]
```diff
@@ -19,4 +19,53 @@
...
+
+
+#:def TIC(n=10)
+  #:global BENCHMARK_NREPS
+  #:set BENCHMARK_NREPS = n
+  block
+    use, intrinsic :: iso_fortran_env, only: int64, real64
+    integer(int64) :: benchmark_i
+
+    integer(int64) :: nb_periodes_initial, nb_periodes_final&
```
Comment (wyphan) [inline src/common/common.fypp:43]: I would put the comma on line 43, before the `&`, but I guess that's just a matter of preference.

Comment (jvdp1) [reply src/common/common.fypp:43]: It is just an habit of mine.
E.g.
```fortran
integer :: i &
              , j &
              , k
```
So if I want to remove `j`, I just have to comment the line with `j` without changing other lines.

**Comment by jvdp1** [comment]
> LGTM. What variable type for `args` do you pass into `TOC`?

A `real64` scalar.

File: src/common/common.fypp (hunk 1 above)
Comment (wyphan) [reply src/common/common.fypp:43]: That's nifty!

**Pull request merged**